bin/shorebird.ps1

```diff
@@ -0,0 +1,95 @@
+# This is the windows equivalent of the `third_party/flutter/bin/internal/shared.sh` script
+# compiles `shorebird_cli/bin/shorebird_cli.dart` to `bin/cache/shorebird.shapshot`
+
+# We are running from $shorebirdRootDir\bin
+$shorebirdBinDir = (Get-Item $PSScriptRoot).FullName
+$shorebirdRootDir = (Get-Item $shorebirdBinDir\..\).FullName
+$shorebirdCacheDir = [IO.Path]::Combine($shorebirdRootDir, "bin", "cache")
+$shorebirdCliDir = [IO.Path]::Combine($shorebirdRootDir, "packages", "shorebird_cli")
+$snapshotPath = [IO.Path]::Combine($shorebirdCacheDir, "shorebird.snapshot")
+$stampPath = [IO.Path]::Combine($shorebirdCacheDir, "shorebird.stamp")
+$flutterPath = [IO.Path]::Combine($shorebirdCacheDir, "flutter")
+$flutter = [IO.Path]::Combine($shorebirdCacheDir, "flutter", "bin", "flutter.bat")
+$shorebirdScript = [IO.Path]::Combine($shorebirdCliDir, "bin", "shorebird.dart")
+$dart = [IO.Path]::Combine($flutterPath, "bin", "cache", "dart-sdk", "bin", "dart.exe")
+
+function Update-Flutter {
+    if (!(Test-Path $flutterPath)) {
+        Write-Output "Cloning flutter, this may take a bit..."
+        git clone --filter=tree:0 https://github.com/shorebirdtech/flutter.git --no-checkout "$flutterPath" *> $null
+    }
+    else {
+        git -C "$flutterPath" fetch *> $null
+    }
+
+    $flutterVersion = Get-Content "$shorebirdBinDir\internal\flutter.version"
+
+    # -c to avoid printing a warning about being in a detached head state.
+    git -C "$flutterPath" -c advice.detachedHead=false checkout "$flutterVersion" *> $null
+
+    # Set FLUTTER_STORAGE_BASE_URL=https://download.shorebird.dev and execute
+    # a `flutter` command to trigger a download of Dart, etc.
+    $env:FLUTTER_STORAGE_BASE_URL = 'https://download.shorebird.dev';
```

Comment:
I wish there was a way to do this in one line, but I wasn't able to find one.  I don't know if this will leak the env out of this script or not?

Reply:
I'm not sure of a more elegant way to do this, but it doesn't seem to change the env of the powershell instance used to invoke the command.